src/olympia/api/middleware.py

```diff
@@ -19,3 +19,26 @@ def process_response(self, request, response):
 
     def process_exception(self, request, exception):
         self.identify_request(request)
+
+
+class APICacheControlMiddleware:
+    def __init__(self, get_response):
+        self.get_response = get_response
+
+    def __call__(self, request):
+        response = self.get_response(request)
+        request_conditions = (
+            request.is_api
+            and request.method in ('GET', 'HEAD')
+            and 'HTTP_AUTHORIZATION' not in request.META
+            and 'disable_caching' not in request.GET
+        )
+        response_conditions = (
+            not response.cookies
+            and response.status_code >= 200
+            and response.status_code < 400
+            and get_max_age(response) is None
```

Comment:
`get_max_age()`  would return a number of seconds if it's set - which can be 0. It will be `None` if it isn't set, which is what we want to check for: we don't want to override if it was explicitly set already by the view.

Reply:
Do we override it anywhere currently?  Or are you just planning for the possibility we might want to in the future?

Reply:
Language Tools and Collection endpoint both set it - they use django's `cache_page` decorator which both caches in memcached _and_ sets `Cache-Control`, with a `max-age` of 24 hours and 1 hour respectively.